specs/experimental/gov-delegation.md

```diff
@@ -0,0 +1,449 @@
+the function MUST calculate and update the voting power of the delegations of `_from` and `_to`. This step MUST apply
+the voting power adjustments to the checkpoints of the `_from`, `_to`, and delegatee addresses.
+
+#### `migrateAccounts`
+
+Migrates the delegation state of the given accounts from the `GovernanceToken` to the `GovernanceDelegation` contract. This
+function MUST iterate over the list of `_accounts` addresses and apply the logic specified in the
+[Migration](#migration) section.
+
+```solidity
+function migrateAccounts(address[] calldata _accounts) external
+```
+
+#### `_delegate`
+
+Delegates voting power from the `_delegator` to the specified delegates in `_newDelegations`. This function MUST
+override any existing delegations of `_delegator`.
+
+```solidity
+function _delegate(address _delegator, Delegation[] memory _newDelegations) internal virtual
+```
+
+The `_delegate` function MUST first check that the length of `_newDelegations` does not exceed `MAX_DELEGATIONS`.
+Then, it MUST calculate the voting power of all active delegations of the `_delegator`, and calculate the voting
+power adjustments for the new delegatee set using `_newDelegations`. Afterwards, the function MUST update
+the voting power of the current and new set of delegatees.
+
+The function MUST then sort the new delegatee set in descending order, and store the delegations as part of the
+`_delegations` mapping. While sorting, the function MUST check for the validity of the delegations, as specified
+in the [Delegation Validation](#delegation-validation) section. Additionally, the function MUST emit a
+`DelegateVotesChanged` event for each change in delegatee voting power.
+
+### Getters
+
+For backwards compatibility, the `GovernanceDelegation` contract MUST implement all public getter functions of the
+`GovernanceToken` related to delegation and voting power. These functions MUST be used by the
+`GovernanceToken` when an account has been migrated to the `GovernanceDelegation` contract. Otherwise,
+the `GovernanceToken` MUST use its own state. Similarly, all of the `GovernanceDelegation` getter functions
+MUST use the `GovernanceToken` state if the account has not been migrated.
+
+#### `checkpoints`
+
+Retrieves the checkpoints for a given user address.
+
+```solidity
+function checkpoints(address _account) external view returns (ERC20Votes.Checkpoint[] memory)
+```
+
+#### `numCheckpoints`
+
+Retrieves the number checkpoints for a given user address.
+
+```solidity
+function numCheckpoints(address _account) external view returns (uint32)
+```
+
+#### `delegates`
+
+Retrieves the delegations of a given user address sorted in descending order by voting power.
+This function is intended to be used by the `GovernanceToken` contract to maximize for backwards compatibility.
+
+```solidity
+function delegates(address _account) external view returns (Delegation[] memory)
```

Comment:
Technically not backwards compatible, for the sake of compatibility, maybe we make this match the original function signature and introduce a new function that returns the full array? Just in the case that someone wants to point their old code at the Alligator and still have it work.